express-api/src/typeorm/Migrations/1717629146741-MoveMetadataToTables.ts

```diff
@@ -0,0 +1,165 @@
+import { In, MigrationInterface, QueryRunner } from 'typeorm';
+import { TimestampType } from '../Entities/TimestampType';
+import { ProjectStatus } from '../Entities/ProjectStatus';
+import { NoteType } from '../Entities/NoteType';
+import { MonetaryType } from '../Entities/MonetaryType';
+import { Task } from '../Entities/Task';
+import { User } from '../Entities/User';
+
+export class MoveMetadataToTables1717629146741 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    const systemUser = await queryRunner.manager.findOne(User, { where: { Username: 'system' } });
+    const systemId = systemUser.Id;
+    const result = await queryRunner.manager.save(ProjectStatus, {
+      Id: 44, //For some reason there is no sequence on project_status
+      Name: 'Close Out',
+      Code: 'CLS-OUT',
+      Description: 'Status related to filling out the close out form.',
+      IsMilestone: true,
+      IsTerminal: true,
+      Route: '/place/holder',
+      CreatedById: systemId,
+    });
+    await queryRunner.manager.insert(TimestampType, [
+      { Name: 'AdjustedOn', IsOptional: true, StatusId: result.Id, CreatedById: systemId }, //ProjectCloseOut
+      { Name: 'ApprovedForSplOn', IsOptional: true, StatusId: 21, CreatedById: systemId }, //ProjectSPLApproval, ProjectERPComplete
+      {
+        Name: 'ClearanceNotificationSentOn',
+        IsOptional: true,
+        StatusId: 14,
```

Comment:
This happens when it leaves ERP? Not sure it belongs with Approved for ERP status then. Almost like we'd need to set it when entering the next status.

Reply:
I guess this works if the frontend allows you to enter a value, change the status, and still save the value you previously entered. There's a few like this.

Reply:
I'm not sure what the next closest status would be. Maybe "In ERP"?